### PR TITLE
refactor: reimplement failurechecker with failure messages

### LINKED ISSUE #4067: [failurechecker] An informative message upon release failure
Feedback from: @ansh0l 

Recent spanner release failed: https://github.com/googleapis/java-spanner/issues/1948

- It didn't give Kokoro job link to troubleshoot. A message to check "prod:cloud-devrel/client-libraries/java/java-spanner/release/stage" would be nice.
- It should be marked as P1 or higher to handle the release failure quickly

### PR DESCRIPTION
**Pull request by chingor13**
This is a major refactor of the code to make it easier to follow and to be able to return the reason for a failure. Since each publication job is different, we can only report on the general expected state.

We also switch to use `@google-automations/issue-utils` to help manage the issue

Fixes #4067

### CODE REVIEW

File: packages/failurechecker/src/failurechecker.ts [hunk 1]
```diff
@@ -51,148 +60,208 @@ export const TimeMethods = {
...
+        } labels = ${releasePullRequest.labels.join(',')}`
+      );
+      failedReleases.push({
+        number: releasePullRequest.number,
+        reason: 'The release job failed -- check the build log.',
+      });
+    }
+
+    // Look for in-progress releases - these usually failed to start the next
+    // step in the pipeline, or failed to report status back
```
Comment (tmatsuo) [inline packages/failurechecker/src/failurechecker.ts:139]: Question: How do we make sure failures are really failures, not a real in-progress?

Comment (chingor13) [reply packages/failurechecker/src/failurechecker.ts:139]: There's a check for the minimum time (3 hours). If the PR is newer than 3 hours, we ignore.

Comment (tmatsuo) [reply packages/failurechecker/src/failurechecker.ts:139]: Ah ok, i would add that in the comment

Comment (chingor13) [reply packages/failurechecker/src/failurechecker.ts]: Added to the comment.

**Review by tmatsuo — APPROVED** [review]

**Pull request merged**